applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_mass_and_damping.h

```diff
@@ -0,0 +1,658 @@
+        ModelPart& rModelPart,
+        TSystemVectorType& rb) override
+    {
+        KRATOS_TRY
+
+    	Timer::Start("BuildRHS");
+
+        BuildRHSNoDirichlet(pScheme, rModelPart, rb);
+
+        //NOTE: dofs are assumed to be numbered consecutively in the BlockBuilderAndSolver
+        block_for_each(BaseType::mDofSet, [&](Dof<double>& r_dof) {
+            const std::size_t i = r_dof.EquationId();
+
+            if (r_dof.IsFixed())
+                rb[i] = 0.0;
+            });
+
+        Timer::Stop("BuildRHS");
+
+        KRATOS_CATCH("")
+    }
+
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const override
+    {
+        Parameters default_parameters = Parameters(R"(
+        {
+            "name"                                 : "block_builder_and_solver_with_mass_and_damping",
+            "block_builder"                        : true,
+            "diagonal_values_for_dirichlet_dofs"   : "use_max_diagonal",
+            "silent_warnings"                      : false
+        })");
+
+        // Getting base class default parameters
+        const Parameters base_default_parameters = BaseType::GetDefaultParameters();
+        default_parameters.RecursivelyAddMissingParameters(base_default_parameters);
+        return default_parameters;
+    }
+
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name()
+    {
+        return "block_builder_and_solver_with_mass_and_damping";
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedBlockBuilderAndSolverWithMassAndDamping";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    TSystemMatrixType mMassMatrix;
+    TSystemMatrixType mDampingMatrix;
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    void GetFirstAndSecondDerivativeVector(TSystemVectorType& rFirstDerivativeVector, TSystemVectorType& rSecondDerivativeVector, ModelPart& rModelPart)
+    {
+        NodesArrayType& r_nodes = rModelPart.Nodes();
+        const int n_nodes = static_cast<int>(r_nodes.size());
```

Comment:
Avoid repeating the type:
```suggestion
        const auto n_nodes = static_cast<int>(r_nodes.size());
```